pkg/controller/service/service_controller.go

```diff
@@ -676,7 +686,19 @@ func (s *ServiceController) lockedUpdateLoadBalancerHosts(service *v1.Service, h
 	// This operation doesn't normally take very long (and happens pretty often), so we only record the final event
 	err := s.balancer.UpdateLoadBalancer(s.clusterName, service, hosts)
 	if err == nil {
-		s.eventRecorder.Event(service, v1.EventTypeNormal, "UpdatedLoadBalancer", "Updated load balancer with new hosts")
+		// If there are no available nodes for LoadBalancer service, we need to update service status.
+		if len(hosts) == 0 {
+			previousState := v1helper.LoadBalancerStatusDeepCopy(&service.Status.LoadBalancer)
+			emptyState := &v1.LoadBalancerStatus{}
```

Comment:
Why should we wipe out the `status.Ingress` field? LoadBalancer might be unavailable when there is no nodes, but presumably it still occupy the external/internal IP.  By removing this field user may assume such IP has been released, which is not true.

Reply:
Ok, so can I just add an "UnAvailableLoadBalancer" event?
Or do you have some better ideas?

Reply:
Don't have any better idea ATM, maybe just add the event for now?

Reply:
Ok.